File: core/src/main/java/pcb/uwu/unit/base/ElectricCurrentUnit.java
package pcb.uwu.unit.base;

import pcb.uwu.core.BaseUnit;
import pcb.uwu.core.BigDecimalAmount;
import pcb.uwu.core.UnitCounter;

import java.util.function.Function;

public abstract class ElectricCurrentUnit extends BaseUnit {

	private final UnitCounter unitCounter;

	public ElectricCurrentUnit(String symbol, String singularName, String pluralName,
			Function<BigDecimalAmount, BigDecimalAmount> translationToCanonical,
			Function<BigDecimalAmount, BigDecimalAmount> translationFromCanonical) {

		super(4, symbol, singularName, pluralName, translationToCanonical, translationFromCanonical);

		unitCounter = new UnitCounter(this);
	}

	@Override
	public Class<? extends BaseUnit> getBaseUnitType() {
		return ElectricCurrentUnit.class;
	}

	@Override
	public UnitCounter getUnitCounter() {
		return unitCounter;
	}
}

File: core/src/main/java/pcb/uwu/unit/base/ElectricCurrentUnit.kt
package pcb.uwu.unit.base

import pcb.uwu.core.BaseUnit
import pcb.uwu.core.BigDecimalAmount
import pcb.uwu.core.UnitCounter
import java.util.function.Function

open class ElectricCurrentUnit(symbol: String,
                               singularName: String,
                               pluralName: String,
                               translationToCanonical: Function<BigDecimalAmount, BigDecimalAmount>,
                               translationFromCanonical: Function<BigDecimalAmount, BigDecimalAmount>)
    : BaseUnit(4, symbol, singularName, pluralName, translationToCanonical, translationFromCanonical)
{
    override val unitCounter
        get() = UnitCounter(this)
    override val baseUnitType = ElectricCurrentUnit::class.java
}